OSDK.toml

```diff
@@ -1,3 +1,7 @@
+[project]
+type = "kernel"
```

Comment:
Cargo only allows one library per crate (but multiple binaries are fine). And we are using lib for kernel. Seems that we cannot have multiple kernels per crate.

Also, the "name" field is a redundancy since they are recorded in the package spec in Cargo manifest.

And I am thinking reserving some designs for modules or components. That's the rational for the current "type" field.

Reply:
But I don't like `[project]` either. Maybe think a better name, or just "type" at root.